accounts/abi/bind/lib.go

```diff
@@ -0,0 +1,152 @@
+package bind
+
+import (
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/event"
+)
+
+func DeployContract2(opts *TransactOpts, bytecode []byte, input []byte, backend ContractBackend) (common.Address, *types.Transaction, error) {
```

Comment:
Maybe we can make DeployContract call DeployContract2?
Something like this:
```diff
@@ -131,17 +130,12 @@ func NewBoundContract(address common.Address, abi abi.ABI, caller ContractCaller
 func DeployContract(opts *TransactOpts, abi abi.ABI, bytecode []byte, backend ContractBackend, params ...interface{}) (common.Address, *types.Transaction, *BoundContract, error) {
        // Otherwise try to deploy the contract
        c := NewBoundContract(common.Address{}, abi, backend, backend, backend)
-
        input, err := c.abi.Pack("", params...)
        if err != nil {
                return common.Address{}, nil, nil, err
        }
-       tx, err := c.transact(opts, nil, append(bytecode, input...))
-       if err != nil {
-               return common.Address{}, nil, nil, err
-       }
-       c.address = crypto.CreateAddress(opts.From, tx.Nonce())
-       return c.address, tx, c, nil
+       addr, tx, err := DeployContract2(opts, bytecode, input, backend)
+       return addr, tx, c, err
 }
 ```